packages/cosmic-swingset/lib/ag-solo/vats/observable.js

```diff
@@ -3,39 +3,23 @@ export default function makeObservablePurse(E, purse, onFulfilled) {
     getName() {
       return E(purse).getName();
     },
-    getAssay() {
-      return E(purse).getAssay();
+    getCurrentAmount() {
+      return E(purse).getCurrentAmount();
     },
-    getBalance() {
-      return E(purse).getBalance();
+    getAllegedBrand() {
+      return E(purse).getAllegedBrand();
     },
-    depositExactly(...args) {
+    deposit(args) {
```

Comment:
Should this (and the three instances below) be `...args` instead of `args`?  That allows for proper variadic arguments.

Reply:
`purse.deposit()` doesn't take varargs, so I don't see how that could function. Am I missing something?

Reply:
Varargs do not need to be supported on the callee.  They become normal arguments.

```js
function callSomething(...args) { something(...args) }
function something(arg1, arg2, arg3) { console.log('something', arg1, arg2, arg3) }
callSomething(); // something undefined undefined undefined
callSomething(1); // something 1 undefined undefined
callSomething(1, 2); // something 1 2 undefined
callSomething(1, 2, 3); // something 1 2 3
callSomething(1, 2, 3, 4);  // something 1 2 3
```

So, `callSomething` perfectly transmits arguments to `something`, where they are handled exactly as though `something` was called directly with those arguments.

Reply:
I get that they'll be passed through. But `purse.deposit()` doesn't seem to be prepared for it's first argument to be a list. Are you saying that something will automatically cause the deposit() method in issuer to iterate over the arguments? Observable isn't doing any iteration.

Reply:
```js
purse.deposit(...args)
```
will convert the array of args to individual arguments.

Reply:
In the new ERTP, `deposit` takes 2 arguments. A problem may arise in the `.deposit(args)` line below (l.14) because this line means that only the first argument is passed from the observable purse to the actual purse.
This could lead to something doing `observablePurse.deposit(payment, halfAmount)` resulting in `actualPurse.deposit(payment)`. This would deposit the entire payment while the intention was to only deposit half

One generic way to fix this is to use spread arguments as @michaelfig suggests:

```js
deposit(...args) {
  return E(purse).deposit(...args)
}
```
Another way is to pass all arguments manually:
```js
deposit(payment, amount) {
  return E(purse).deposit(payment, amount)
}
```